src/react/components/pages/editorPage/editorPage.tsx

```diff
@@ -325,6 +327,30 @@ export default class EditorPage extends React.Component<IEditorPageProps, IEdito
         });
     }
 
+    private async updateProjectTagsFromAsset(asset: IAssetMetadata) {
+        const assetTags = new Set();
+        asset.regions.forEach((region) => region.tags.forEach((tag) => assetTags.add(tag.name)));
+
+        const newTags: ITag[] = this.props.project.tags ? this.props.project.tags.slice() : [];
+        let updateTags = false;
+
+        assetTags.forEach((tag) => {
+            if (!this.props.project.tags || this.props.project.tags.length === 0 ||
+                this.props.project.tags.filter((projectTag) => tag === projectTag.name).length === 0 ) {
```

Comment:
I would use `tags.find(...)` here.  It stops at the first found element vs. traversing the whole array.